### PR TITLE
Set an explicit request for bazel jobs.

### PR DESCRIPTION
**Pull request by spxtr**
Also run `line` jobs not on build nodes.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/1154)

### CODE REVIEW

**Comment by spxtr** [comment]
#1152

File: prow/jobs.yaml [hunk 1]
```diff
@@ -67,6 +67,9 @@ kubernetes/kubernetes:
       # Bazel needs privileged mode in order to sandbox builds.
       securityContext:
         privileged: true
+      resources:
+        requests:
+          cpu: 6
```
Comment (mikedanese) [inline prow/jobs.yaml]: Can you add a limit as well?

Comment (spxtr) [inline prow/jobs.yaml]: What should it be?

Comment (spxtr) [inline prow/jobs.yaml]: Also, do I need memory request/limits? The docs are kinda confusing.

Comment (mikedanese) [inline prow/jobs.yaml]: set it to 6 as well. memory would be good to do too.

File: prow/kube/types.go [hunk 2]
```diff
@@ -153,6 +154,16 @@ type ConfigMapKeySelector struct {
 	Key  string `json:"key,omitempty"`
 }
 
+type Resources struct {
+	Requests *ResourceRequest `json:"requests,omitempty"`
+	Limits   *ResourceRequest `json:"limits,omitempty"`
+}
+
+type ResourceRequest struct {
```
Comment (spxtr) [inline prow/kube/types.go]: But yes, this package does predate client-go.

File: prow/jobs.yaml (hunk 1 above)
Comment (mikedanese) [inline prow/jobs.yaml]: Can you give 10G of memory?

Comment (mikedanese) [inline prow/jobs.yaml]: both limit and request

Comment (spxtr) [inline prow/jobs.yaml]: Done.

File: prow/kube/types.go (hunk 2 above)
Comment (spxtr) [inline prow/kube/types.go]: Code simplicity, git repo size, and binary size, in descending order. Go is excellent for writing simple REST clients, and rolling our own lets us fine-tune without writing a wrapper package.

File: prow/jobs.yaml [hunk 3]
```diff
@@ -67,6 +67,13 @@ kubernetes/kubernetes:
...
+          memory: 10G
+        limits:
+          cpu: 6
```
Comment (mikedanese) [inline prow/jobs.yaml]: I wonder if 6 needs to be quoted. Also Gi is the binary SI version G is decimal. Probably doesn't matter

**Comment by mikedanese** [comment]
LGTM

**Pull request merged**